view/frontend/web/js/view/payment/method-renderer/omise-offline-paynow-method.js

```diff
@@ -0,0 +1,52 @@
+define(
+    [
+        'ko',
+        'Omise_Payment/js/view/payment/omise-base-method-renderer',
+        'Magento_Checkout/js/view/payment/default',
+        'Magento_Checkout/js/model/full-screen-loader',
+        'Magento_Checkout/js/action/redirect-on-success',
+        'Magento_Checkout/js/model/quote',
+        'Magento_Checkout/js/model/error-processor'
+    ],
+    function (
+        ko,
+        Base,
+        Component,
+        fullScreenLoader,
+        redirectOnSuccessAction,
+        quote
+    ) {
+        'use strict';
+
+        return Component.extend(Base).extend({
+            defaults: {
+                template: 'Omise_Payment/payment/offline-paynow-form'
+            },
+
+            isPlaceOrderActionAllowed: ko.observable(quote.billingAddress() != null),
+
+            code: 'omise_offline_paynow',
+            restrictedToCurrencies: ['sgd'],
+
+            /**
+             * Hook the placeOrder function.
+             * Original source: placeOrder(data, event); @ module-checkout/view/frontend/web/js/view/payment/default.js
+             *
+             * @return {boolean}
+             */
+            placeOrder: function(data, event) {
```

Comment:
Is this the same as another 'offline' method? Should we not be extending from a generic 'offline' method?

Reply:
then it requires refactoring of all offline payment methods. should be done in next pr, by person who will take it over.

Reply:
Well, the same was already being done with 'offsite' methods. A pattern had been established. As soon as additional offline methods appeared, the same pattern should have been followed instead of cut and pasting. Should be easy to do right now

Reply:
Looks like the convenience store method on this branch is not using ANY of the reworked JS code at all. Is this right?

Reply:
i.e The other offline methods are now fixed, but this one seems to be using old code style

Reply:
yes, because place order function is different.

Reply:
That's no reason to totally ignore the rest of the structure changes, and the structure in place is like a class/subclass structure - you can easily override a parent method. This JS stuff is all documented in the Magento documentation - that's where I got the refactoring idea in the first place

Reply:
ok, it will be done in another pr by somebody who will take it over.

Reply:
I have 2 prs, that needs to be merged with this PR, correcting admin panel texts, and grouping stuff in admin panel.

Reply:
Sorry, it should be done now. All other payment methods are following this structure